pkg/server/image_pull.go

```diff
@@ -274,9 +280,7 @@ func (c *criService) getResolver(ctx context.Context, ref string, cred func(stri
 		}
 		// Continue to try next endpoint
 	}
-	client := &http.Client{
-		CheckRedirect: checkRedirecter(ctx),
-	}
+	client.CheckRedirect = checkRedirecter(ctx)
```

Comment:
I feel like we might as well move this up above to simplify the client construction logic. I can't think of any reason we wouldn't want logging for the configured mirrors as well.

Reply:
I didn't look into why this was originally only added to the fallback client but I'll add this. The commit msg was already updated on the last push but I'll fix up the title too :)